### PR TITLE
[tune] add output flag for Tune CLI

### PR DESCRIPTION
**Pull request by andrewztan**
## What do these changes do?

Adds output flag (-o / --output) to ls / lsx commands for Tune CLI. Builds on top of #3983

### CODE REVIEW

**Comment by andrewztan** [comment]
@richardliaw

**Comment by hartikainen** [comment]
Would it be better to output these in a csv instead of pkl? Considering that the data being outputted does not need to have python objects saved, i.e. the data is tabular, I think it would be better serialize the things with something else than pickle.

File: python/ray/tune/commands.py [hunk 1]
```diff
@@ -160,16 +162,24 @@ def list_trials(experiment_path,
 
     print_format_output(checkpoints_df)
 
+    if output:
+        experiment_path = os.path.expanduser(experiment_path)
+        output_filepath = os.path.join(experiment_path, output + ".pkl")
```
Comment (hartikainen) [inline python/ray/tune/commands.py]: We should probably expect the `output` to include the files extension.

Comment (hartikainen) [reply python/ray/tune/commands.py]: You could then also implicitly support csv or pickle based on the extension.

File: python/ray/tune/commands.py [hunk 2]
```diff
@@ -160,16 +162,27 @@ def list_trials(experiment_path,
 
     print_format_output(checkpoints_df)
 
+    if output:
+        experiment_path = os.path.expanduser(experiment_path)
+        output_path = os.path.join(experiment_path, output)
+        if output.split('.')[-1].lower() in ['p', 'pkl', 'pickle']:
+            checkpoints_df.to_pickle(output_path)
+        else:
```
Comment (hartikainen) [inline python/ray/tune/commands.py]:
```suggestion
        elif output.split('.')[-1].lower() == 'csv':
```

File: python/ray/tune/commands.py [hunk 3]
```diff
@@ -160,16 +162,27 @@ def list_trials(experiment_path,
...
+            checkpoints_df.to_csv(output_path, index=False)
```
Comment (hartikainen) [inline python/ray/tune/commands.py]:
```suggestion
            checkpoints_df.to_csv(output_path, index=False)
        else:
            raise ValueError("Unsupported filetype: {}".format(output))
```

Comment (hartikainen) [reply python/ray/tune/commands.py]: Same below

**Comment by andrewztan** [comment]
@richardliaw @hartikainen ready for review / merge

File: python/ray/tune/commands.py [hunk 4]
```diff
@@ -233,6 +248,16 @@ def list_experiments(project_path,
 
     print_format_output(info_df)
 
+    if output:
+        output_path = os.path.join(base, output)
+        if output.split('.')[-1].lower() in ["p", "pkl", "pickle"]:
```
Comment (hartikainen) [inline python/ray/tune/commands.py]: You could actually extract the file extension using the `os.path.splitext`:
```python
file_extension = os.path.splitext(output)[1]
if file_extension in ("p", "pkl", "pickle"):
    ...
elif file_extension == "csv":
    ...
else:
    ...
```

Comment (andrewztan) [reply python/ray/tune/commands.py]: that's helpful, thanks!

**Review by hartikainen — APPROVED** [review]

**Comment by hartikainen** [comment]
Looks good!

File: python/ray/tune/commands.py [hunk 5]
```diff
@@ -162,16 +164,30 @@ def list_trials(experiment_path,
...
     print_format_output(checkpoints_df)
 
+    if output:
+        experiment_path = os.path.expanduser(experiment_path)
+        output_path = os.path.join(experiment_path, output)
+        file_extension = os.path.splitext(output)[1].lower()
+        if file_extension in (".p", ".pkl", ".pickle"):
+            checkpoints_df.to_pickle(output_path)
+        elif file_extension == ".csv":
+            checkpoints_df.to_csv(output_path, index=False)
```
Comment (richardliaw) [inline python/ray/tune/commands.py]: can this be a helper function that you reuse in both ls, lsx calls?

Comment (richardliaw) [reply python/ray/tune/commands.py]: Actually, maybe we don't need to do this in this PR and might just be overabstracting

**Comment by richardliaw** [comment]
We should add this to the docs.

**Pull request merged**